### ISSUE TITLE
Error: write EPIPE in AbstractIngestionEngine / Promise { <pending> }

### ISSUE THREAD

**Issue by chriszs** [tvkitchen/tv-kitchen#59]
#### Current Behavior

When I try to inherit from the AbstractIngestionEngine and use it to stream from a file (after fixing #57 and #58), I get:

```
events.js:292
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at WriteWrap.onWriteComplete [as oncomplete] (internal/stream_base_commons.js:92:16)
Emitted 'error' event on Socket instance at:
    at Socket.onerror (_stream_readable.js:753:14)
    at Socket.emit (events.js:315:20)
    at Socket.EventEmitter.emit (domain.js:485:12)
    at emitErrorNT (internal/streams/destroy.js:100:8)
    at emitErrorCloseNT (internal/streams/destroy.js:68:3)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  errno: -32,
  code: 'EPIPE',
  syscall: 'write'
}
```

After adding a log statement for error events for each stream in the pipe, I get `Promise { <pending> }` as an error from the `ingestPayload` stream. This is the promise produced by `this.producer.send` and passed as the first argument to `done`, which is interpreted as an error because that's convention for callbacks.

#### Input
```sh
yarn babel-node src/components/ingestion/test.js
```

```js
// src/components/ingestion/test.js
import 'module-alias/register'
import AbstractIngestionEngine from './AbstractIngestionEngine'

class FileIngestionEngine extends AbstractIngestionEngine {
	constructor(path) {
		super()

		this.path = path
	}

	path = null

	getInputStream = () => fs.createReadStream(this.path)
}

const engine = new FileIngestionEngine(__dirname + '/in.ts')

engine.start()
```

#### Expected Behavior
I expected it to not throw that error.

#### Possible Solutions
Wait for the promise from sending to the producer, call `done()` without a first argument on success.
```
.then(result => done(null,result)).catch(done)
```

#### Related Issues
#57 and #58

**Comment by slifty** [comment]
Let the record show that I very much do not like that callback convention.

![image](https://user-images.githubusercontent.com/208884/84107928-f66dfa00-a9ec-11ea-9716-7856f02d5fcc.png)

**Comment by chriszs** [comment]
Would you prefer a more `async/await` style? Is this my chance to sneak my promisification into this?

**Comment by slifty** [comment]
Oh boy yes let's do it!

**Comment by slifty** [comment]
Closed by #56
